test/integration/consumer_group_test.go

```diff
@@ -31,87 +37,270 @@ func TestConsumerGroup(t *testing.T) {
 	ctx := context.Background()
 	ns, cleaner := helpers.Setup(ctx, t, env)
 
+	d, s, i, p := deployMinimalSvcWithKeyAuth(ctx, t, ns.Name)
+	cleaner.Add(d)
+	cleaner.Add(s)
+	cleaner.Add(i)
+	cleaner.Add(p)
+
+	addedHeader := header{
+		K: "X-Test-Header",
+		V: "added",
+	}
+	// Use the same header key as plugin name.
+	pluginRespTrans := configurePlugin(
+		ctx, t, ns.Name, "response-transformer-advanced-1", "response-transformer-advanced", fmt.Sprintf(
+			`{
+				"add": {
+					"headers": [
+						"%s: %s"
+					]
+				}
+			 }`,
+			addedHeader.K, addedHeader.V,
+		),
+	)
+	cleaner.Add(pluginRespTrans)
+
+	const rateLimitValue = 100
+	pluginRateLimit := configurePlugin(
+		ctx, t, ns.Name, "rate-limiting-advanced-1", "rate-limiting-advanced", fmt.Sprintf(
+			`{
+				"limit": [%d],
+				"window_size": [100],
+				"namespace": "test",
+				"sync_rate": -1,
+				"window_type": "fixed"
+			 }`,
+			rateLimitValue,
+		),
+	)
+	cleaner.Add(pluginRateLimit)
+
+	// Currently Gateway supports only one plugin per consumer group, read more
+	// https://konghq.atlassian.net/browse/FTI-5282, but it silently accepts many
+	// but it runs only one of them. So we have to be careful.
```

Comment:
I was thinking of adding a validation rule to the admission webhook to catch CGs with multiple plugins then, but if that's going to be fixed in a patch release of Kong 3.4, I think that's ok. I created an issue so we have this tracked somewhere.
```suggestion
	// Currently Gateway supports only one plugin per consumer group, read more
	// https://konghq.atlassian.net/browse/FTI-5282, but it silently accepts many
	// but it runs only one of them. So we have to be careful.
	// https://github.com/Kong/kubernetes-ingress-controller/issues/4472 tracks extending 
	// this test once the limitation on Kong side is fixed.
```

Reply:
Perfect, thank you!